docs/deployment-gcp.md

```diff
@@ -0,0 +1,134 @@
+# Deploying Canopy on Google Cloud Platform (GCP) - Step-by-Step Guide
+
+
+## Introduction
+In this guide, we'll walk through the process of deploying Canopy on Google Cloud Platform (`GCP`). 
+The steps include setting up GCP, creating a Docker repository, pulling and tagging the `Canopy` image, and finally deploying it using Google Cloud Run.
+
+## Prerequisites
+Before you begin, make sure you have the following installed:
+
+- [Docker](https://docs.docker.com/engine/install/)
+- [Google Cloud SDK](https://cloud.google.com/sdk/docs/install)
+
+and make sure you have a project set up in `GCP`.
+
+## Step 1: Install and Authenticate Google Cloud SDK
+Open your terminal and run the following commands:
+
+```bash
+# Authenticate with your Google Cloud account
+gcloud auth login
+
+# Set the GCP project
+gcloud config set project {project-name}
+```
+
+## Step 2: Create a Docker Repository on GCP (Optional)
```

Comment:
Shouldn't we just push the image ourselves to a GCP repository that we'll set to be public?  
Then people won't need to pull from github and re-push to their own GCP repository.

Or is that not supported somehow?

Reply:
We can push to GCP also but then we will have no one source of truth. DockerHub is supported by both but I don't think we should change where we push based solely on the enforcement of the cloud provider. I checked a service similar to Cloud Run in GCP in AWS. It enforces AWS Registry so that means we also need to push to AWS Registry. I see two options, either just pushing to one place or pushing to all main cloud providers.